src/test/java/org/jenkinsci/plugins/workflow/cps/SnippetizerTest.java

```diff
@@ -308,6 +310,12 @@ public void monomorphicListSymbol() throws Exception {
     }
 
     @Test
+    public void escapeCharacters() throws Exception {
+        DisplaynameWithEscapeCharState displaynameWithEscapeCharState = new DisplaynameWithEscapeCharState(new DisplaynameEscapeCharData("one"));
+        st.assertRoundTrip(displaynameWithEscapeCharState, "displaynameWithEscapeCharState([firstArg: 'one'])");
```

Comment:
`assertRoundTrip` does not exercise either of the `dsld` or `gdsl` views AFAICT. Did you try running this test with your code patches temporarily reverted to see if it would fail?